### PR TITLE
vte3: fix build for Linux

### PR DESCRIPTION
**Pull request by cho-m**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

Also try making `vala` build-time-only dependency.

https://github.com/Homebrew/homebrew-core/runs/3063445949?check_suite_focus=true
```
../meson.build:241:2: ERROR: Assert failed: needs G++ >= 7 for C++17 support
```

### CODE REVIEW

File: Formula/vte3.rb
Comment (danielnachun) [inline Formula/vte3.rb:28]: Based on the build failure we probably need to add `linux-headers` - it's looking for function which should be defined there.

Comment (cho-m) [reply Formula/vte3.rb:28]: I tried searching for `TIOCGPTPEER` in `linux-headers`, but it isn't available there:
```console
# grep -R TIOCGPT /home/linuxbrew/.linuxbrew/opt/linux-headers/include/*
/home/linuxbrew/.linuxbrew/opt/linux-headers/include/asm-generic/ioctls.h:#define TIOCGPTN	_IOR('T', 0x30, unsigned int) /* Get Pty Number (of pty-mux device) */
/home/linuxbrew/.linuxbrew/opt/linux-headers/include/asm-generic/ioctls.h:#define TIOCGPTLCK	_IOR('T', 0x39, int) /* Get Pty lock state */
```

Based on https://man7.org/linux/man-pages/man4/tty_ioctl.4.html, it mentioned `TIOCGPTPEER` should be available since Linux 4.13

Homebrew's `linux-headers` is v4.4.80.

Comment (cho-m) [reply Formula/vte3.rb:28]: I guess there was a previous attempt to update `linux-headers` to v5 #69081

Alternatively, there was plans to keep this aligned with docker image (is it 5.10.25?):
https://github.com/Homebrew/homebrew-core/pull/64211#issuecomment-723656422

Probably blocked until we update that.

Comment (danielnachun) [reply Formula/vte3.rb:28]: We will have to make versioned formula for newer headers.  @sjackman suggested we use version 4.15 and make a formula called `linux-headers@4.15` and also add an alias `linux-headers@4.4` for the current `linux-headers` formula.

Comment (danielnachun) [reply Formula/vte3.rb:28]: We will need this for `qemu` as well, so this will probably help with several formulae.

Comment (cho-m) [reply Formula/vte3.rb:28]: Is there a reason we need to keep v4.4 as the main formula rather than using a newer version?

v4.15 is already EOL.
v4.14 (EOL 01/2024) or v4.19 (EOL 12/2024) would probably be better.

Or jumping on the v5 train like v5.4 (EOL 2025)  or 5.10 (EOL 2026).
Though, v5 would probably make things less compatible.

Comment (danielnachun) [reply Formula/vte3.rb:28]: I think 4.15 was suggested because that is version that comes with Ubuntu 18.04 LTS, which will be upgrading to after the core merge is done.  Your point about using an LTS version as defined by the kernel upstream (not Ubuntu) is well taken though.  I think @sjackman should really weigh in on this - I don't have a strong opinion either way on whether we should go with Ubuntu LTS or upstream kernel LTS.

Comment (cho-m) [reply Formula/vte3.rb:28]: Okay, I think I see reason.
Ubuntu 16.04 uses 4.4.0 as main package
Ubuntu 18.04 uses 4.15.0 as main package.

Comment (cho-m) [reply Formula/vte3.rb:28]: @danielnachun I see that it makes sense to align to whatever OS version bottles are built on, i.e. Ubuntu's `linux-headers-generic`

We can go with that and maybe swap names/aliases when moving to 18.04 LTS.

Comment (danielnachun) [reply Formula/vte3.rb:28]: Yeah that's what we were planning.  Feel free to create a new formula for the 4.15 headers, or let me know if you'd like me to do it.  We can also try to bottle `qemu` after that is merged, assuming its dependencies are bottled as well.

Comment (cho-m) [reply Formula/vte3.rb:28]: You can go ahead.

Comment (danielnachun) [reply Formula/vte3.rb:28]: I've submitted the a new formula for `linux-headers@4.15`.  It passed CI fine so as soon as it is approved, merged and bottled, we can rebase this and add it as a dependency.

**Review by SMillerDev — APPROVED** [review]

**Review by danielnachun — APPROVED** [review]

**Pull request merged**
